File: Packages/CaptureKit/Sources/CaptureKit/CaptureGeometry.swift
import CoreGraphics

public enum CaptureGeometry {
    /// Convert a rect in Cocoa global coordinates (bottom-left origin, points)
    /// into a top-left-origin pixel rect relative to a captured display image.
    public static func pixelRect(forGlobalRect rect: CGRect,
                                 inDisplayFrame display: CGRect,
                                 scale: CGFloat) -> CGRect {
        let xLocal = (rect.minX - display.minX) * scale
        let yTopLocal = (display.maxY - rect.maxY) * scale
        return CGRect(x: xLocal, y: yTopLocal,
                      width: rect.width * scale, height: rect.height * scale)
    }
}

File: windows/src/BetterScreenshot.Capture/CaptureGeometry.cs
using BetterScreenshot.Core;

namespace BetterScreenshot.Capture;

/// <summary>
/// Converts a selection rectangle (top-left, logical coords within a display) into a top-left pixel rect
/// relative to that display's captured image. Windows is top-left origin everywhere, so — unlike the macOS
/// version — there is no Y-flip.
/// </summary>
public static class CaptureGeometry
{
    public static PxRect PixelRect(PxRect selection, PxRect display, double scale) =>
        new((selection.X - display.X) * scale,
            (selection.Y - display.Y) * scale,
            selection.Width * scale,
            selection.Height * scale);
}
